### PR TITLE
[Addon] vela core shard manager

### PR DESCRIPTION
**Pull request by Somefive**
Signed-off-by: Yin Da <yd219913@alibaba-inc.com>

### Checklist

I have:

- [x] Title of the PR starts with type (e.g. `[Addon]` , `[example]` or `[Doc]`).
- [ ] Updated/Added any relevant [documentation](https://kubevela.io/docs/reference/addons/overview) and [examples](https://github.com/kubevela/catalog/tree/master/examples).
- [ ] New addon should be put in [experimental](https://github.com/kubevela/catalog/tree/master/experimental/addons).
- [ ] Update addon should modify the `version` in `metadata.yaml` to generate a new version.

####  Verified Addon promotion rules

If this pr wants to promote an experimental addon to verified, you must check whether meet these conditions too:
  - [ ] This addon must be tested by addon's [e2e-test](./test/e2e-test/addon-test) to guarantee this addon can be enabled successfully.
  - This addon must have some basic but necessary information.
    - [ ] An accessible icon url and source url defined in addon's `metadata.yaml`.
    - [ ] A detail introduction include a basic example about how to use and what's the benefit of this addon in `README.md`.
    - [ ] Also provide an introduction in KubeVela [documentation](https://kubevela.net/docs/reference/addons/overview).
    - [ ] It's more likely to be accepted if useful examples provided in example [dir](examples/).

### CODE REVIEW

File: addons/vela-core-shard-manager/parameter.cue [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+	// +usage=The namespace of the vela-prism to be installed
+	namespace: *"vela-system" | string
+
+	// vela-core replica parameters
+
+	deploymentName: *"kubevela-vela-core" | string
+	containerName: *"kubevela" | string
+	instanceName: *"kubevela-slave" | string
+
+	nShards: *1 | int
```
Comment (wonderflow) [inline addons/vela-core-shard-manager/parameter.cue:21]:
```suggestion
	nShards: *1 | int | >0
```

File: addons/vela-core-shard-manager/README.md [hunk 2]
```diff
@@ -0,0 +1,10 @@
+# vela-core-shard-manager
+
+The vela-core-shard-manager is a manager for slave application controllers when sharding is enabled.
+
+## install
+
+```shell
+vela addon enable vela-core-shard-manager nShards=3
+```
+
```
Comment (wonderflow) [inline addons/vela-core-shard-manager/README.md:10]:
```suggestion
## Why using addon for sharding?
```

Please explain the benifits

Comment (wonderflow) [reply addons/vela-core-shard-manager/README.md:10]: Concerns:

1. does the addon need  special permission?
2. the stability and security, will the addon be wrongly removed and affected the core?

Comment (Somefive) [reply addons/vela-core-shard-manager/README.md:10]: Done

File: addons/vela-core-shard-manager/parameter.cue (hunk 1 above)
Comment (Somefive) [reply addons/vela-core-shard-manager/parameter.cue]: Done

**Review by wonderflow — DISMISSED** [review]

File: addons/vela-core-shard-manager/parameter.cue [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+	// +usage=The name of the addon application
+	name: "addon-vela-core-shard-manager"
+}
+
+parameter: {
+
+	// global parameters
+
+	// +usage=The namespace of the vela-prism to be installed
+	namespace: *"vela-system" | string
```
Comment (wangyikewxgm) [inline addons/vela-core-shard-manager/parameter.cue:13]: Do we truly need this paramter?

Comment (Somefive) [reply addons/vela-core-shard-manager/parameter.cue]: Removed.

**Review by wangyikewxgm — APPROVED** [review]
LGTM

**Review by wangyikewxgm — APPROVED** [review]

**Review by wangyikewxgm — APPROVED** [review]

**Review by barnettZQG — APPROVED** [review]
LGTM

**Pull request merged**